### DISCUSSION TITLE
Can get TFs with tf2_ros.TransformListener

### DISCUSSION THREAD

**Discussion by erodios** [gstavrinos/ai_msc_robotic_systems_assignments#8] [Q&A] [answered]
So for the second essay i am trying to get the TFs of the possitions that the robot should go by creating a TransformListener. The problem that i have is that when i try to store the TF into a variable, without checking if the TF is available, my node crashes and the only message I get is "Transition Failed" (for the on_activate lifecycle). After this I tried to wait until the TF is available with the can_transform function but it gets stuck there. I have used as time the following 
[ (self.get_clock().now() ),   (self.get_clock().now() - Duration(seconds=5))].  I have tried this with the parameter "use_sim_time" True and False, I have also used a timeout Duration in the can_transform function, with a value up to 120 sec and it still didn't manage to find the TF. The TF that I am currently trying to get is map->tt_table_center_link.

**Comment by gstavrinos** [comment, marked as answer]
Have you tried something like this?

`self.tf_buffer.can_transform("map", "tt_table_link",  rclpy.time.Time()))`

Short explanation of why this could potentially work:
The `rclpy.time.Time() ` is basically the same as a time `t=0.0`, which is the equivalent of requesting the "latest tf in the tree".

Also, always keep in my that TSs need to "spin" in order to be updated, thus it is crucial to not be "trapped" in non-main threads (like callbacks) when creating buffers/listeners and when executing lookups.

Keep me updated on how it went! Good luck!

**Reply by erodios** [reply]
Well thanks a lot! I tried the rclpy.time.Time() right after i started the discussion but it didn't work. The spin hint helped though.
